### DISCUSSION TITLE
Consumer batch mode

### DISCUSSION THREAD

**Discussion by hinxx** [edenhill/librdkafka#3631] [Q&A] [answered]
Going through the `rdkafka.h` I noticed that `rd_kafka_consume_batch()` is considered legacy. Is there a similar non-legacy method in improving consumer performance?
On a related note, there seems to be quite a number of functions marked as legacy. How should I interpret the 'legacy' in such cases?

**Comment by edenhill** [comment, marked as answer]
The standard consumer_poll() API is typically fast enough, the bottleneck is typically fetching messages from Kafka, not returning the messages to the application.
Are you seeing significant performance improvements with consume_batch() over consumer_poll() (et.al)?

Deprecated means two things:
 * we're no longer putting our efforts into these APIs.
 * the API will probably be removed if we ever do a major version bump (nothing is planned).

**Reply by hinxx** [reply]
Thanks for clarifying!
I'm not that far down the line of being able to report any performance differences as I'm still in a process of coding up the (first iteration of) data consumption pipeline.
I was defaulting to `rd_kafka_consumer_poll()` until I noticed a whole set of functions that might be more performant (as per the comments in the header file).
